sstables/sstables.cc

```diff
@@ -2432,7 +2433,16 @@ input_stream<char> sstable::data_stream(uint64_t pos, size_t len,
                 pos, len, std::move(options), permit);
         }
     }
-
+    if (_components->checksum && raw == raw_stream::no) {
```

Comment:
Does this keep the meaning of raw_stream? In the previous block, raw_stream::yes means compressed  data, raw_stream::no means decompressed data. Here it means checksum not validated / yes validated. That's not consistent.

Reply:
I also expect we need lifecycle management of the checksum data structure via lw_shared_ptr. Consider two overlapping validations (or other uses of the checksum) of the same sstable.

Reply:
>Does this keep the meaning of raw_stream? In the previous block, raw_stream::yes means compressed data, raw_stream::no means decompressed data. Here it means checksum not validated / yes validated. That's not consistent.

I was thinking of re-purposing this to indicate a request for an input stream with a plain file data source. No higher-level logic. But since I also introduce the `integrity_check` option, it's not necessary. I will revert this change.

>I also expect we need lifecycle management of the checksum data structure via lw_shared_ptr. Consider two overlapping validations (or other uses of the checksum) of the same sstable.

I made the checksum component shareable and weakly referencable in a later patch. In case of uncompressed SSTables, `sstable::data_stream()` always gets called with a reference being held by the caller.

Reply:
Please fold the changes, we don't introduce a bug and then fix it in the same series.